### ISSUE TITLE
Create basic dashboards for our signals

### ISSUE THREAD

**Issue by ryantibs** [cmu-delphi/covidcast#32]
In writing the Facebook blog post, I wanted to know a bunch of basic things about the Facebook survey, and found myself writing code to find the answers.  Then I thought we should just build a very basic dashboard so that we can constantly check these things.  Dashboard is probably too generous, I'm just talking about a notebook that gets run automatically each day.  I put one up [here](https://cmu-delphi.github.io/covidcast/R-notebooks/fb_dashboard.html).

This was pretty easy to build with the functionality in our covidcast R package.  If people agree it's useful, then we should make one of these for each of our signals, and write a script that runs these each morning.

Note: I guess we should discuss what this is for and who would use it.

* I think it's mainly for us, and it allows us to keep basic tabs on our indicators just by browsing a notebook.
* We could assign one person from the engineering team to take a look at one of these notebooks, once per day (which ideally would update itself).
* For example, I can see from the notebook linked above that the Facebook survey has been consistently trending down in responses per day and counties per day, over the last month ... if this keeps up we might go and tell Facebook.
* We could also use it to do some very basic anomaly detection, nothing fancy---we'd have to add a few plots for this.  This might be similar to what we're already getting from Larry (Pickett?) but could be worth just implementing something very basic on our side as well. 

Bottom line, I would say, is that: nothing fancy here.  I think we want to keep these notebooks simple to maintain and simple to look at.

Tagging krivard capnrefsmmat jsharpna because they might have ideas of who could tackle this. It's relatively low priority but would be a good way for somebody to onboard themselves both with respect to getting to know our R client and our signals.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1198968030437421/1198970455381965) by [Unito](https://www.unito.io/learn-more)

**Comment by benjaminysmith** [comment]
I am adding basic coverage and sampling dashboards for all of the signals, similar to what we already have for symptom surveys.

- [ ] Doctor Visits
- [ ] Google Health Trends
- [ ] Hospital Admissions
- [ ] Indicator Combination
- [ ] JHU Cases and Deaths
- [ ] Quidel
- [ ] SafeGraph Mobility
- [ ] Symptom Surveys
- [ ] USAFacts Cases and Deaths

**Comment by benjaminysmith** [comment]
Ideas from @sangwon-hyun as extension to basic coverage:

- In the coverage graph, it might be helpful to visualize more than a snapshot e.g. which counties we used to have data in, but no longer do.

It could be helpful to put a basic anomaly detection on these two graphs:
- Whether the most recent 30 days are trending down, according to a best fit line's slope.
- Whether there is an abrupt drop (2 or 3 standard deviations?) compared to the most recent 10 day average?

**Comment by ryantibs** [comment]
@benjaminysmith Thanks for the continued progress here.  I took a look at https://delphi.midas.cs.cmu.edu/~krivard/covidcast/ and I like the way things are shaping up!

(Nb. on #240, I renamed the dashboard Rmd files in order and change the make.R file just a bit, so that it'll only compile Rmd files with "dashboard" in them.  To exclude other Rmd files in the R-notebooks directory like the covidhub eval notebook being featured on that PR.  This shouldn't be causing any conflicts I don't believe, just a heads up.)

Some comments: 
- I really like the heat maps of coverage over the last 28 days, I think there's some good info being shared there.
- Can we underlay in a different color (say, could use with pink as in the other heat maps) counties that *ever* appear in the signal?  This way we can see whether, in the last 28 days, we're missing a large batch of counties that appeared at some previous point in time.
- I would recommend being as systematic/consistent as we can with what we show for all the different signals.  Basically, make the same comparisons and keep the same structure as much as you can.  This will help make the human process of flipping through the different dashboards as lightweight as possible (because we'd know what to expect).
- I would also recommend structuring it like this: 3 tabs (see the notebook in #240 for example code on how to setup tabs, it's very simple) named:
    i. Counties
    ii. Metro areas
    iii. Sample size
- The "Counties" tab shows the coverage over time graph, and the coverage map.  (When counties aren't available you can just keep this tab, and say "Not available.")
- The "Metro areas" tab does the same, but for Metro areas.
- The "Sample size" tab shows the sample size over time graph.  (When sample size isn't available, you can just keep this tab, and say "Not available.")
- I don't think we need states if we have these two (counties and metro areas).
- Contrary to what I said before, you can nix the correlations from any of the individual signal dashboards.
- In the correlations dashboard, you can again structure it to use tabs, to break it up and make it more readable.
- Finally, we can combine all of these dashboards very conveniently into one single html file, using bookdown.  See [here](https://stackoverflow.com/questions/25824795/how-to-combine-two-rmarkdown-rmd-files-into-a-single-output/51521542#51521542) for a minimal example of how to combine Rmds using bookdown.

What do you think?  I imagine everything here is pretty straightforward but just will take time to setup (particularly bookdown if you've never used it before).  Maybe @sangwon-hyun @nloliveira @sarah-colq can help?  You can divide up tasks in a modular way?

**Comment by benjaminysmith** [comment]
Thanks for the feedback Ryan. I'm running these at http://dashboards-prod-01.delphi.cmu.edu/ (internal only for now) too, with a nicer HTML wrapper (and the not-yet checked in ones in https://github.com/cmu-delphi/covidcast/pull/235).

These are great ideas for V2. I'll think about how to split this up effectively and update here.

**Comment by benjaminysmith** [comment]
Ok, looking at the requests I think we can chop this into:

1. Add map support for whether a county has *ever* appeared (in addition to the existing 28 day maps)
2. Create tabbed view for county/metros/samples (and refactor code to be able to generate for each of these easily)
3. Combine into single HTML
4. Create tabbed correlations dashboard
5. Add simple anomaly detection graphing -- e.g., recent points out of band (I think this is more complax and should wait for the AD indicator work to wrap up so we can share)

**Comment by benjaminysmith** [comment]
Unfortunately we can't (easily) use tabs if we use bookdown: https://github.com/rstudio/bookdown/issues/393

Leaning toward keeping the tabs and continuing to use a custom HTML wrapper instead. @ryantibs if you prefer ditching the tabs in favor of bookdown as the container we can do that too.

**Comment by ryantibs** [comment]
@benjaminysmith Thanks for the continued improvements to the dashboards!  I'm in favor of doing whatever is the lightest weight (simplest) while maintaining ease-of-navigation.  So whether that's tabs or anything else, don't have a strong opinion.

(E.g., a table-of-contents could be just as easy to navigate as tabs, and may jell better with bookdown?)

**Comment by benjaminysmith** [comment]
Sounds good to me, thanks.
